New login view

mariaAcevedo opened the pull request:
### Summary
* New login frontend
### Screenshot
![image](https://user-images.githubusercontent.com/51001022/58722270-6777dc00-83ad-11e9-8007-7b7db71bc673.png)

### Trello Cards
https://trello.com/c/8uXwUZod/16-u1-frontend-login-componentes-b%C3%A1sicos
https://trello.com/c/S1DCi9U0/17-u1-frontend-login-validaciones

mariaAcevedo committed:
New login view

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/ILoginView.java:
```diff
@@ -0,0 +1,25 @@
+package ar.com.wolox.android.example.ui.login;
+/**
```
Please remove all commented code.

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/ILoginView.java:
```diff
@@ -0,0 +1,25 @@
+*/
+
+/**
+ * Comment
```
Add a meaningful message to this comment.

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java:
```diff
@@ -0,0 +1,183 @@
+     */
+    @Override
+    public void setListeners() {
+        //sacar esto
```
we should remove something ?

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java:
```diff
@@ -0,0 +1,68 @@
+
+    }
+
+    void storeUsername(String text) {
```
If we are not going to use this method, we can remove it.

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java:
```diff
@@ -0,0 +1,68 @@
+        */
+        SharedPreferences sharedPref = loginActivity.getSharedPreferences("UserSession", Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putString("Mail", mail);
```
We should create a constant with the code nsame  **"mail"**

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java:
```diff
@@ -0,0 +1,68 @@
+        SharedPreferences sharedPref = loginActivity.getSharedPreferences("UserSession", Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putString("Mail", mail);
+        editor.putString("Password", password);
```
Same here

DemianIac commented on app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java:
```diff
@@ -0,0 +1,68 @@
+        editor.putString("Mail", mail);
+        editor.putString("Password", password);
+        editor.commit();
+        getView().onSignUpFinished("Estas registrado");
```
The message should be handled by the View, in this case de Fragment. You can send a code Message (An int or something like that.) or you can use onSingUpFinished when the user is logged and onError(message) when the login fails.

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+        android:layout_marginStart="16dp"
+        android:layout_marginEnd="16dp"
+        android:gravity="center"
+        android:hint="Mail"
```
We should move hardcoded strings to the resource Strings.xml

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+
+    <ImageView
+        android:id="@+id/imageViewFoto"
+        android:layout_width="413dp"
```
I think that this photo should match the parent.  (you can use match_parent or change it in the editor)

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+    <ImageView
+        android:id="@+id/imageViewFoto"
+        android:layout_width="413dp"
+        android:layout_height="208dp"
```
Maybe we can use a guideline to define the size of the photo.

You can read more here: https://riggaroo.co.za/constraintlayout-guidelines-barriers-chains-groups/

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+
+    <ImageView
+        android:id="@+id/imageViewLogo"
+        android:layout_width="140dp"
```
We should set a size in the Dimens.xml file.

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+    <ImageView
+        android:id="@+id/imageViewLogo"
+        android:layout_width="140dp"
+        android:layout_height="108dp"
```
Same here. Maybe set a more appropriate number too, like 120.

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="20dp"
+        android:layout_marginEnd="16dp"
+        android:background="#2196F3"
```
Colors can be defined in the resource Colors.xml

DemianIac commented on app/src/main/res/layout/fragment_example.xml:
```diff
@@ -1,45 +1,111 @@
+        android:layout_marginTop="20dp"
+        android:layout_marginEnd="16dp"
+        android:background="#2196F3"
+        android:text="SIGN UP"
```
Same with text --> Strings.xml.

DemianIac reviewed (changes requested):
Nice work!.
Some comments:

- Remove all commented code, or code that your are not going to use
- Try to use the resources files when needed (Strings.xml, Colors.xml, Dimes.xml)
- All the code and comments should be english 
- All the code that you are going to change in the future can be tagged with //TODO : message

Let me know if you need help with the view!.
😄

mariaAcevedo committed:
Corrections

mariaAcevedo committed:
Corrections made:

All comments were erased
MAil and Password constant were created for LOginPresenter class
Fragment display was corrected to match the orinal one.
Meaning full comments were written on LoginFragment and LoginACtivity

mariaAcevedo committed:
Gradient added to the photo, constant added and added message error as a resource

DemianIac reviewed (approved):
![giphy (1)](https://user-images.githubusercontent.com/20775059/58903770-91e2d580-86dc-11e9-8914-a11d546bf432.gif)
